theme/static/js/hs_file_browser.js

```diff
@@ -78,332 +101,376 @@ function formatBytes(bytes) {
     else return(bytes / 1073741824).toFixed(1) + " GB";
 }
 
-// Updates the state of toolbar and menu buttons when a selection is made
+// Initial state for actions where they can appear.
+var initActionState = {
+    disabled: false, // When an item is disabled, it is disabled everywhere
+    /*-- Used to control where and when the action is visible --*/
+    // The main toolbar in the file browser
+    toolbar: {
+        hidden: false,
+    },
+    // The menu that shows up when right clicking on a file or folder
+    fileMenu: {
+        hidden: false,
+    },
+    // The menu that shows up when clicking on the main container. Typically contains directory specific actions
+    containerMenu: {
+        hidden: false,
+    }
+};
+
+/*
+    Updates the state of toolbar and menu buttons when a selection is made.
+    All items start as visible and enabled by default. Performs checks and disables as it goes.
+*/
 function updateSelectionMenuContext() {
     var selected = $("#fb-files-container li.ui-selected");
     var resourceType = $("#resource-type").val();
     var mode = $("#hs-file-browser").attr("data-mode");
     var selectedFileName = selected.children(".fb-file-name").text().toUpperCase();
-    var flagDisableOpen = false;
-    var flagDisableDownload = false;
-    var flagDisableRename = false;
-    var flagDisablePaste = false;
-    var flagDisableZip = false;
-    var flagDisableUnzip = false;
-    var flagDisableCut = false;
-    var flagDisableDelete = false;
-    var flagDisableSetGenericFileType = false;
-    var flagDisableSetFileSetFileType = false;
-    var flagDisableSetGeoRasterFileType = false;
-    var flagDisableSetNetCDFFileType = false;
-    var flagDisableSetGeoFeatureFileType = false;
-    var flagDisableSetRefTimeseriesFileType = false;
-    var flagDisableSetTimeseriesFileType = false;
-    var flagDisableRemoveAggregation = false;
-    var flagDisableGetLink = false;
-    var flagDisableCreateFolderOrFile = false;
+
+    // correspond to 'data-fb-action' attribute values in the file browser controls that you wish to manage
+    const uiActions = [
+        "createFolder",
+        "cut",
+        "delete",
+        "download",
+        "downloadZipped",
+        "getLink",
+        "getRefUrl",
+        "open",
+        "paste",
+        "removeAggregation",
+        "rename",
+        "setFileSetFileType",
+        "setGenericFileType",
+        "setGeoFeatureFileType",
+        "setGeoRasterFileType",
+        "setNetCDFFileType",
+        "setRefTimeseriesFileType",
+        "setTimeseriesFileType",
+        "subMenuSetContentType",
+        "unzip",
+        "updateRefUrl",
+        "zip",
+    ];
+    var uiActionStates = {};
+
+    // Initialize the action states
+    $.each(uiActions, function (i, val) {
+        uiActionStates[val] = $.extend(true, {}, initActionState);  // Deep copy
```

Comment:
uiActionStates is an improvement from the flag variables that existed before 👍

Reply:
I agree. A dictionary makes them easy to find.